app/src/main/java/org/simple/clinic/forgotpin/confirmpin/ForgotPinConfirmPinScreenController.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.forgotpin.confirmpin
+
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.ObservableTransformer
+import io.reactivex.rxkotlin.ofType
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.facility.FacilityRepository
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.widgets.ScreenCreated
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
+
+typealias Ui = ForgotPinConfirmPinScreen
+typealias UiChange = (Ui) -> Unit
+
+class ForgotPinConfirmPinScreenController @Inject constructor(
+    private val userSession: UserSession,
+    private val facilityRepository: FacilityRepository
+) : ObservableTransformer<UiEvent, UiChange> {
+
+  override fun apply(upstream: Observable<UiEvent>): ObservableSource<UiChange> {
+    val replayedEvents = upstream.compose(ReportAnalyticsEvents()).replay().refCount()
+
+    return Observable.merge(
+        showUserNameOnScreenStarted(replayedEvents),
+        showFacilityOnScreenCreated(replayedEvents),
+        openFacilityChangeScreen(replayedEvents)
+    )
+  }
+
+  private fun showUserNameOnScreenStarted(events: Observable<UiEvent>): Observable<UiChange> {
+    return events.ofType<ScreenCreated>()
+        .flatMap { userSession.requireLoggedInUser() }
+        .map { user -> { ui: Ui -> ui.showUserName(user.fullName) } }
+  }
+
+  private fun showFacilityOnScreenCreated(events: Observable<UiEvent>): Observable<UiChange> {
+    return events.ofType<ScreenCreated>()
+        .flatMap { userSession.requireLoggedInUser() }
+        .switchMap { facilityRepository.currentFacility(it) }
+        .map { facility -> { ui: Ui -> ui.showFacility(facility.name) } }
```

Comment:
Have you stopped using `it` now?

Reply:
Depends. I don't have a hard or fast rule. If the lambda seems cleaner and non-obvious with the name, I use it. Otherwise, I stick with `it`. It's all very subjective.